From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Thu, 1 Apr 2021 21:46:17 +0200
Subject: Configure: allow to enable ktls if target does not start with Linux

The Debian build system uses a `debian' target which sets CFLAGS and
then we have for instance debian-amd64 which inherits from
linux-x86_64 and debian. So far so good.

Since the target name does not start with `linux', the build system does not
enable ktls. So in order to get enabled, I
added a
        `enable           => [ "ktls" ],'
to the generic linux config which sets it explicit). Having this set, we can
check for it instead matching the target name.

This commit is based on changes for afalgeng in commit
   9e381e8a01859 ("Configure: allow to enable afalgeng if target does not start with Linux")

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 Configurations/10-main.conf | 2 +-
 Configure                   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index f5e5754b3a6c..b27a60cf3923 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -695,7 +695,7 @@ my %targets = (
         shared_target    => "linux-shared",
         shared_cflag     => "-fPIC",
         shared_ldflag    => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
-        enable           => [ "afalgeng" ],
+        enable           => [ "afalgeng", "ktls" ],
     },
     "linux-generic64" => {
         inherit_from     => [ "linux-generic32" ],
diff --git a/Configure b/Configure
index 613b48e7d9bf..70afd617f8e1 100755
--- a/Configure
+++ b/Configure
@@ -1687,7 +1687,7 @@ unless ($disabled{devcryptoeng}) {
 
 unless ($disabled{ktls}) {
     $config{ktls}="";
-    if ($target =~ m/^linux/) {
+    if (grep { $_ eq 'afalgeng' } @{$target{enable}}) {
         my $usr = "/usr/$config{cross_compile_prefix}";
         chop($usr);
         if ($config{cross_compile_prefix} eq "") {
